janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java

```diff
@@ -1582,4 +1583,27 @@ public final boolean isClosed() {
     public boolean hasModifications() {
         return !addedRelations.isEmpty() || !deletedRelations.isEmpty();
     }
+
+    @Override
+    public void expireSchemaElement(final long id) {
+        if (vertexCache.contains(id)) {
+            final InternalVertex v = vertexCache.get(id, externalVertexRetriever);
+            if (v instanceof JanusGraphSchemaVertex) {
+                JanusGraphSchemaVertex sv = (JanusGraphSchemaVertex) v;
+                sv.resetCache();
+                if (sv.getDefinition().containsKey(TypeDefinitionCategory.INTERNAL_INDEX)
```

Comment:
I would prefer calling `sv.asIndexType().isMixedIndex()` which is shorter and does the same thing.

Reply:
Thanks for the review again @li-boxuan!  I've changed the code to use `sv.asIndexType().isMixedIndex()` as you requested.

I've also removed the try-catch as you and @porunov requested.  It was added to handle a possible NPE thrown by `getIndexTransaction`.  Instead, I've added a method `hasIndexTransaction` to avoid the NPE.

Reply:
Actually, I think I still need a check that the `JanusGraphSchemaVertex` is indeed an index type.

Reply:
> I think I still need a check that the JanusGraphSchemaVertex is indeed an index type.

I don't think that is necessary but there is no harm to do so.